rfcs/20200817-jit-code-generation-for-aarch64/README.md

```diff
@@ -0,0 +1,256 @@
+`translator/indluce/xbyak_translator_for_aarch64` directory contains files 
+that implement the main functions of the translation, such as 
+`decodeAndTransToAArch64` and functions called by `decodeAndTransToAArch64`. 
+The rules for translating x64 instructions to AArch64 ones are implemented 
+in files in `translator/indluce/xbyak_translator_for_aarch64/instructions` 
+directory. At the moment, `Xbyak_translator_aarch64` supports 200 
+instructions that are used when running oneDNN on a CPU which can execute 
+AVX512/AVX2/AVX/SSE4.1 instructions.
+
+### Adding a dependency on xed at the build stage:
+
+To build oneDNN with Xbyak_translator_aarch64, it must be assumed that 
+a pre-build xed library is available, built from the xed sources. 
+Additional CMake modules will be needed, such as `FindXED.cmake` and 
+`XED.cmake` which will only impact AArch64 builds. 
+[We modified just a few lines](https://github.com/fujitsu/xed/commit/92ee2830ca48fed76baa0a36e7d6ef8a3afe5741) 
+of the original xed sources and disclosed as 
+[fujitsu/xed](https://github.com/fujitsu/xed/tree/fjdev "fujitsu/xed"). 
+
+### Adding primitive entry to src/cpu/cpu_\*_list.cpp
+
+As JIT code for AArch64 is ready to be generated at this point, primitives are 
+added to `src/cpu/cpu_*_list.cpp` as follows not only for convolution but also 
+eltwise, batch_normalization, pooling, etc. This proposal is the same as the 
+proposal in 
+[#795](https://github.com/oneapi-src/oneDNN/pull/795) and the implementation 
+for x64.
+
+```
+static const std::map<conv_impl_key_t, std::vector<pd_create_f>> impl_list_map {
+    // FWD fp
+    {{forward, f32, f32, f32}, {
+        CPU_INSTANCE_X64(jit_avx512_common_dw_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_common_1x1_convolution_fwd_f32_t)
+        CPU_INSTANCE_X64(jit_avx512_core_f32_wino_conv_2x3_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_core_f32_wino_conv_4x3_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_common_convolution_winograd_fwd_t)
+        CPU_INSTANCE_X64(jit_avx512_common_convolution_fwd_t<f32>)
+        CPU_INSTANCE_X64(jit_avx2_dw_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_avx2_1x1_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_sse41_dw_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_sse41_1x1_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_avx2_convolution_fwd_t)
+        CPU_INSTANCE_X64(jit_sse41_convolution_fwd_t)
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_dw_convolution_fwd_t)         <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_1x1_convolution_fwd_f32_t)    <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_core_f32_wino_conv_2x3_fwd_t)        <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_core_f32_wino_conv_4x3_fwd_t)        <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_convolution_winograd_fwd_t)   <- Added
+        CPU_INSTANCE_AARCH64(jit_aarch64_sve_512_common_convolution_fwd_t<f32>)       <- Added
```

Comment:
really minor nit-picking: `aarch64_` part of the class name could be dropped, as it is already guarded by `CPU_INSTANCE_AARCH64()` and will live in the corresponding namespace.

Reply:
Thank you. It's LGTM.